src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareSaveScmpOperation.cs

```diff
@@ -0,0 +1,137 @@
+            this.OperationId = Guid.NewGuid().ToString();
+        }
+
+
+        public void Execute(TaskExecutionMode mode = TaskExecutionMode.Execute)
+        {
+            if (this.CancellationToken.IsCancellationRequested)
+            {
+                throw new OperationCanceledException(this.CancellationToken);
+            }
+
+            try
+            {
+                SchemaCompareEndpoint sourceEndpoint = SchemaCompareUtils.CreateSchemaCompareEndpoint(this.Parameters.SourceEndpointInfo, this.SourceConnectionString);
+                SchemaCompareEndpoint targetEndpoint = SchemaCompareUtils.CreateSchemaCompareEndpoint(this.Parameters.TargetEndpointInfo, this.TargetConnectionString);
+
+                SchemaComparison comparison = new SchemaComparison(sourceEndpoint, targetEndpoint);
+
+                if (Parameters.ExcludedSourceObjects != null)
+                {
+                    foreach (var sourceObj in this.Parameters.ExcludedSourceObjects)
+                    {
+                        SchemaComparisonExcludedObjectId excludedObjId = CreateExcludedObject(sourceObj);
+                        if (excludedObjId != null)
+                        {
+                            comparison.ExcludedSourceObjects.Add(excludedObjId);
+                        }
+                    }
+                }
+
+                if (Parameters.ExcludedTargetObjects != null)
+                {
+                    foreach (var targetObj in this.Parameters.ExcludedTargetObjects)
+                    {
+                        SchemaComparisonExcludedObjectId excludedObjId = CreateExcludedObject(targetObj);
+                        if (excludedObjId != null)
+                        {
+                            comparison.ExcludedTargetObjects.Add(excludedObjId);
+                        }
+                    }
+                }
+
+                if (this.Parameters.DeploymentOptions != null)
+                {
+                    comparison.Options = SchemaCompareUtils.CreateSchemaCompareOptions(this.Parameters.DeploymentOptions);
+                }
+
+                comparison.SaveToFile(this.Parameters.ScmpFilePath, true);
+
+            }
+            catch (Exception e)
+            {
+                ErrorMessage = e.Message;
+                Logger.Write(TraceEventType.Error, string.Format("Schema compare save settings operation {0} failed with exception {1}", this.OperationId, e.Message));
+                throw;
+            }
+        }
+
+        private SchemaComparisonExcludedObjectId CreateExcludedObject(SchemaCompareObjectId sourceObj)
+        {
+            try
+            {
+                ObjectIdentifier id = new ObjectIdentifier(sourceObj.Name.Split("."));
+                SchemaComparisonExcludedObjectId excludedObjId = new SchemaComparisonExcludedObjectId(sourceObj.SqlObjectType, id);
+
+                return excludedObjId;
+            }
+            catch
+            {
+                return null;
```

Comment:
Swallowing exceptions like this makes me nervous. Can we scope to a particular exception?

Reply:
Oh it gets reflection related excptions (not one prticulr type) in creating exclude object. I can log. I am swallowing because I dont want to fail save operation if a exclude object could not be created.

Do you think trowing and failing the save is safer?

Reply:
My bad.. not reflection.. it does a switch in dacfx.. so will only catch argumentException and throw on rest. Thanks for pointing out.